server/src/main/java/io/druid/segment/realtime/firehose/SqlFirehoseFactory.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.druid.segment.realtime.firehose;
+
+import com.fasterxml.jackson.annotation.JacksonInject;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import com.google.common.base.Preconditions;
+import io.druid.data.input.impl.prefetch.PrefetchSqlFirehoseFactory;
+import io.druid.guice.annotations.Smile;
+import io.druid.java.util.common.StringUtils;
+import io.druid.metadata.MetadataStorageConnectorConfig;
+import io.druid.metadata.SQLFirehoseDatabaseConnector;
+import org.skife.jdbi.v2.ResultIterator;
+import org.skife.jdbi.v2.exceptions.CallbackFailedException;
+import org.skife.jdbi.v2.exceptions.ResultSetException;
+import org.skife.jdbi.v2.exceptions.StatementException;
+
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SqlFirehoseFactory extends PrefetchSqlFirehoseFactory<String>
+{
+  @JsonProperty
+  private final List<String> sqls;
+  @JsonProperty
+  private final MetadataStorageConnectorConfig connectorConfig;
+  private final ObjectMapper objectMapper;
+  @JsonProperty
+  private final SQLFirehoseDatabaseConnector sqlFirehoseDatabaseConnector;
+  private final Boolean foldCase;
+
+  @JsonCreator
+  public SqlFirehoseFactory(
+      @JsonProperty("sqls") List<String> sqls,
+      @JsonProperty("maxCacheCapacityBytes") Long maxCacheCapacityBytes,
+      @JsonProperty("maxFetchCapacityBytes") Long maxFetchCapacityBytes,
+      @JsonProperty("prefetchTriggerBytes") Long prefetchTriggerBytes,
+      @JsonProperty("fetchTimeout") Long fetchTimeout,
+      @JsonProperty("foldCase") Boolean foldCase,
```

Comment:
You can use `@JsonProperty("foldCase") boolean foldCase` and it should be false if `foldCase` is not specified in the spec.

Reply:
Fixed.